core/src/main/java/com/linecorp/armeria/server/UncaughtExceptionsServerErrorHandler.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.server.logging.LoggingService;
+
+import io.netty.channel.EventLoopGroup;
+
+/**
+ * Implementation of {@link ServerErrorHandler} that is used to decorate other ServerErrorHandlers.
+ * When services are not annotated with {@link LoggingService}, exceptions are not caught be default.
+ * {@link UncaughtExceptionsServerErrorHandler} is used to catch those uncaught exceptions.
+ */
+public class UncaughtExceptionsServerErrorHandler implements ServerErrorHandler {
```

Comment:
- How about making it package-private? I think a user will not really need to access this class directly.
- How about `ExceptionReportingServerErrorHandler`?

Reply:
Sure, it's for internal use only, so it's better to be package-private. 
The name seems better. 😀